src/sentry/notifications/utils/participants.py

```diff
@@ -85,51 +86,89 @@ def get_participants_for_group(
     return participants_by_provider
 
 
-def get_reason(
-    user: User, value: NotificationSettingOptionValues, user_ids: set[int]
-) -> int | None:
-    # Members who opt into all deploy emails.
-    if value == NotificationSettingOptionValues.ALWAYS:
-        return GroupSubscriptionReason.deploy_setting
-
-    # Members which have been seen in the commit log.
-    elif value == NotificationSettingOptionValues.COMMITTED_ONLY and user.id in user_ids:
-        return GroupSubscriptionReason.committed
-    return None
-
-
 def get_participants_for_release(
-    projects: Iterable[Project], organization: Organization, user_ids: set[int]
+    projects: Iterable[Project],
```

Comment:
@barkbarkimashark why is it we have to modify this function so much? I must be missing something.

Reply:
This method was originally only being called by `DeployNotification`, so the implementation assumed as much. But since we'll be splitting it up `ActiveReleaseSummaryNotification` away from `DeployNotification`, we have to introduce the notification type and pass that into this function.